core/common/src/main/java/alluxio/wire/TieredIdentity.java

```diff
@@ -194,9 +203,37 @@ public boolean equals(Object o) {
       if (!(o instanceof LocalityTier)) {
         return false;
       }
-      LocalityTier that = (LocalityTier) o;
-      return mTierName.equals(that.mTierName)
-          && Objects.equal(mValue, that.mValue);
+      LocalityTier otherTier = (LocalityTier) o;
+      String otherTierName = otherTier.getTierName();
+      if (!mTierName.equals(otherTierName)) {
+        return false;
+      }
+      String otherTierValue = otherTier.getValue();
+      if (mValue == null && otherTierValue == null) {
+        return true;
```

Comment:
hm, on further thought my suggestion for overriding `equals` wasn't great. If the value is null, we don't want locality tiers to be considered equal for the purpose of matching. Otherwise, if a node has identity
"node=null, rack=A" and calls `nearest` on ["node=null, rack=B", "node=null, rack=A"], the node will incorrectly choose "node=null, rack=B" because the nodes "match". However, if we don't know the node names, it's better to continue looking for matches by looking at the racks. 

Perhaps call the method `matches` instead?

Reply:
Done